# CITEMO (A lightweight single-cell multimodal omics analysis framework)
* We propose a lightweight framework, called CITEMO, to analyze single-cell multimodal omics data. 
* This project is a practice of CITEMO, which can inspire other single-cell multi-multi-omics sequencing data analysis. CITEMO.ipynb shows an example of using CITEMO to analyze CBMC multimodal omics samples. All data involved in this project are stored in the Data folder. 
* It is strongly recommended to install the specified version of the python package as shown in Requirement. 
* It is worth noting that running citemo.ipynb in different environments may get different results. You should combine your biological experience and try to constantly adjust the model parameters to obtain the best results.
## Requirement
* Python >= 3.8.0
* jupyterlab >= 3.0.9
* numpy == 1.19.5
* ray == 1.1.0
* matplotlib == 3.3.4
* seaborn == 0.11.1
* scipy == 1.6.1
* scikit-learn == 0.24.1
* statsmodels == 0.12.2
* pandas == 1.2.3
* pillow == 8.1.2
* jedi == 0.18.0
* umap-learn == 0.5.1
* phenograph == 1.5.7
* bokeh == 2.3.0
## Dataset
* CBMC
<https://satijalab.org/seurat/archive/v3.2/multimodal_vignette.html>
* HBMC
<https://satijalab.org/seurat/articles/weighted_nearest_neighbor_analysis.html>
* PBMC1k
<https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.0.0/pbmc_1k_protein_v3>
* PBMC10k
<https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.0.0/pbmc_10k_protein_v3>
* MALT
<https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.0.0/malt_10k_protein_v3>
* PBMC5k
<https://support.10xgenomics.com/single-cell-gene-expression/datasets/3.0.2/5k_pbmc_protein_v3>
